### PR TITLE
feat: stateful

### PR DESCRIPTION
**Pull request by lightsing**
There's some issue when recovering the signature

### CODE REVIEW

**Comment by lispc** [comment]
debugging method:  
    let our revm output the "struct log trace" using https://github.com/paradigmxyz/revm-inspectors/blob/main/src/tracing/builder/geth.rs, and compare with geth side trace.

File: crates/stateful/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+            genesis_config.init_code_db(&mut code_db)?;
+            let zktrie =
+                genesis_config.init_zktrie::<Poseidon, _, _>(&mut zktrie_db, NoCacheHasher)?;
+            history_db.set_block_storage_root(0, *zktrie.root().unwrap_ref())?;
+        }
+
+        let shutdown = Arc::new(AtomicBool::new(false));
+
+        let pipeline_rx = Fetcher::spawn(
+            20,
```
Comment (lispc) [inline crates/stateful/src/lib.rs:82]: 20 rpc fetcher to get blocks?

Comment (lispc) [reply crates/stateful/src/lib.rs:82]: ok yes. you explained in function impl codes.

**Review by lispc — DISMISSED** [review]

**Comment by lispc** [comment]
there seems to be some commented codes.  you can have a check. whether to delete them or keep.

File: crates/primitives/src/lib.rs [hunk 2]
```diff
@@ -203,12 +203,10 @@ pub trait TxTrace {
 
     /// Try to build a typed transaction
     fn try_build_typed_tx(&self) -> Result<TypedTransaction, SignatureError> {
-        let chain_id = self.chain_id();
-
         let tx = match self.ty() {
             0x0 => {
                 let tx = TxLegacy {
```
Comment (i-m-aditya) [inline crates/primitives/src/lib.rs:208]: whats the reason for the change, it is causing EF testcases to fail `transaction validation error` @lightsing

**Pull request merged**